<!DOCTYPE html>
<html class="no-js" lang="en">
<head>
	<meta charset="UTF-8">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<title>Create Phylogenetic Tree of SARS CoV 2 by UShER - OmicX</title>
	<script>(function(d,e){d[e]=d[e].replace("no-js","js");})(document.documentElement,"className");</script>
	<meta name="description" content="">
		<meta property="og:title" content="Create Phylogenetic Tree of SARS CoV 2 by UShER" />
<meta property="og:description" content="There are nearly 14 million viral genome sequences right now in the GISAID EpiCoV ™ database. It is not likely  to infer the phylogenetic relationships for such a huge dataset by traditional maximum likelyhood or Bayesian methods in a shor time period. The UShER package was developed to generate ultra-large phylogenetic tree of SARS-CoV-2 genomes. The algorithm of the UShER program is to place new samples onto an existing phylogeny using maximum parsimony method. It is able to place given SARS-CoV-2 genome sequences into the GISAID global phylogeny in a couple of hours.  This program is particularly helpful in understanding the relationships of newly sequenced SARS-CoV-2 genomes with each other and with previously sequenced genomes in a global phylogeny.
The UShER package is composed by four programs:

UShER: a program that rapidly places new samples onto an existing phylogeny using maximum parsimony.
matUtils:  a toolkit for querying, interpreting and manipulating the mutation-annotated trees (MATs).
matOptimize:  a program to rapidly and effectively optimize a mutation-annotated tree (MAT) for parsimony using subtree pruning and regrafting (SPR) moves within a user-defined radius.
RIPPLES: a program that uses a phylogenomic technique to rapidly and sensitively detect recombinant nodes and their ancestors in a mutation-annotated tree (MAT).

The taxoniumtools and Taxonium website are used to display the MAT generated by UShER." />
<meta property="og:type" content="article" />
<meta property="og:url" content="https://omicx.cc/posts/2022-11-22-create-phylogenetic-tree-of-sars-cov-2-by-usher/" /><meta property="article:section" content="posts" />
<meta property="article:published_time" content="2022-11-22T16:39:19+08:00" />
<meta property="article:modified_time" content="2022-11-22T16:39:19+08:00" />

		<meta itemprop="name" content="Create Phylogenetic Tree of SARS CoV 2 by UShER">
<meta itemprop="description" content="There are nearly 14 million viral genome sequences right now in the GISAID EpiCoV ™ database. It is not likely  to infer the phylogenetic relationships for such a huge dataset by traditional maximum likelyhood or Bayesian methods in a shor time period. The UShER package was developed to generate ultra-large phylogenetic tree of SARS-CoV-2 genomes. The algorithm of the UShER program is to place new samples onto an existing phylogeny using maximum parsimony method. It is able to place given SARS-CoV-2 genome sequences into the GISAID global phylogeny in a couple of hours.  This program is particularly helpful in understanding the relationships of newly sequenced SARS-CoV-2 genomes with each other and with previously sequenced genomes in a global phylogeny.
The UShER package is composed by four programs:

UShER: a program that rapidly places new samples onto an existing phylogeny using maximum parsimony.
matUtils:  a toolkit for querying, interpreting and manipulating the mutation-annotated trees (MATs).
matOptimize:  a program to rapidly and effectively optimize a mutation-annotated tree (MAT) for parsimony using subtree pruning and regrafting (SPR) moves within a user-defined radius.
RIPPLES: a program that uses a phylogenomic technique to rapidly and sensitively detect recombinant nodes and their ancestors in a mutation-annotated tree (MAT).

The taxoniumtools and Taxonium website are used to display the MAT generated by UShER."><meta itemprop="datePublished" content="2022-11-22T16:39:19+08:00" />
<meta itemprop="dateModified" content="2022-11-22T16:39:19+08:00" />
<meta itemprop="wordCount" content="1876">
<meta itemprop="keywords" content="Tree,UShER,Taxonium," />
		<meta name="twitter:card" content="summary"/><meta name="twitter:title" content="Create Phylogenetic Tree of SARS CoV 2 by UShER"/>
<meta name="twitter:description" content="There are nearly 14 million viral genome sequences right now in the GISAID EpiCoV ™ database. It is not likely  to infer the phylogenetic relationships for such a huge dataset by traditional maximum likelyhood or Bayesian methods in a shor time period. The UShER package was developed to generate ultra-large phylogenetic tree of SARS-CoV-2 genomes. The algorithm of the UShER program is to place new samples onto an existing phylogeny using maximum parsimony method. It is able to place given SARS-CoV-2 genome sequences into the GISAID global phylogeny in a couple of hours.  This program is particularly helpful in understanding the relationships of newly sequenced SARS-CoV-2 genomes with each other and with previously sequenced genomes in a global phylogeny.
The UShER package is composed by four programs:

UShER: a program that rapidly places new samples onto an existing phylogeny using maximum parsimony.
matUtils:  a toolkit for querying, interpreting and manipulating the mutation-annotated trees (MATs).
matOptimize:  a program to rapidly and effectively optimize a mutation-annotated tree (MAT) for parsimony using subtree pruning and regrafting (SPR) moves within a user-defined radius.
RIPPLES: a program that uses a phylogenomic technique to rapidly and sensitively detect recombinant nodes and their ancestors in a mutation-annotated tree (MAT).

The taxoniumtools and Taxonium website are used to display the MAT generated by UShER."/>

	<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
	<link rel="dns-prefetch" href="//fonts.googleapis.com">
	<link rel="dns-prefetch" href="//fonts.gstatic.com">
	<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Open+Sans:400,400i,700">

	<link rel="stylesheet" href="/css/style.css">
	

	<link rel="shortcut icon" href="/favicon.ico">

    
    
<script async src="https://www.googletagmanager.com/gtag/js?id=G-3NS8PNT6QP"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-3NS8PNT6QP');
</script>


    
</head>
<body class="body">
	<div class="container container--outer">
		<header class="header">
	<div class="container header__container">
		
	<div class="logo logo--mixed">
		<a class="logo__link" href="/" title="OmicX" rel="home">
			<div class="logo__item logo__imagebox">
					<img class="logo__img" src="/img/logo-512x512.png">
				</div><div class="logo__item logo__text">
					<div class="logo__title">OmicX</div>
					<div class="logo__tagline">Omics &amp; Statistics</div>
				</div>
		</a>
	</div>
		<div class="divider"></div>
	</div>
</header>

		<div class="wrapper flex">
			<div class="primary">
			
<main class="main" role="main">
	<article class="post">
		<header class="post__header">
			<h1 class="post__title">Create Phylogenetic Tree of SARS CoV 2 by UShER</h1>
			<div class="post__meta meta"><div class="meta__item-author meta__item">
	<svg class="meta__icon icon icon-author" width="16" height="16" viewBox="0 0 12 16"><path d="M6 1c2.2 0 3.5 2 3.5 4.5C9.5 7 8.9 8.2 8 9c2.9.8 4 2.5 4 5v1H0v-1c0-2.5 1.1-4.2 4-5-.9-.8-1.5-2-1.5-3.5C2.5 3 3.8 1 6 1z"/></svg><span class="meta__text">omicx.cc</span>
</div>
<div class="meta__item-datetime meta__item">
	<svg class="meta__icon icon icon-time" width="16" height="14" viewBox="0 0 30 28"><path d="M15 0C7 0 1 6 1 14s6 14 14 14 14-6 14-14S23 0 15 0zm0 25C9 25 4 20 4 14S9 3 15 3s11 5 11 11-5 11-11 11zm1-18h-2v8.4l6.8 4.4L22 18l-6-3.8V7z"/></svg><time class="meta__text" datetime="2022-11-22T16:39:19&#43;08:00">November 22, 2022</time></div><div class="meta__item-categories meta__item"><svg class="meta__icon icon icon-category" width="16" height="16" viewBox="0 0 16 16"><path d="m7 2l1 2h8v11h-16v-13z"/></svg><span class="meta__text"><a class="meta__link" href="/categories/phylogenetics/" rel="category">Phylogenetics</a>, <a class="meta__link" href="/categories/software/" rel="category">Software</a>
	</span>
</div></div>
		</header>
		
<div class="post__toc toc">
	<div class="toc__title">Page content</div>
	<div class="toc__menu">
		<nav id="TableOfContents">
  <ul>
    <li><a href="#1-installation">1. Installation</a>
      <ul>
        <li><a href="#11-install-usher-package-via-conda">1.1 Install UShER package via conda</a></li>
        <li><a href="#12-install-taxoniumtools">1.2 Install taxoniumtools</a></li>
      </ul>
    </li>
    <li><a href="#2-procedures">2. Procedures</a>
      <ul>
        <li><a href="#21-download-sequences-and-metadata">2.1 Download sequences and metadata</a></li>
        <li><a href="#22-operation">2.2 Operation</a></li>
        <li><a href="#23-tree-visualization">2.3 Tree visualization</a></li>
      </ul>
    </li>
    <li><a href="#references">References</a></li>
    <li><a href="#appendix-a-command-line-options">Appendix A. Command line options</a>
      <ul>
        <li><a href="#a1-usher-help">A.1 <code>UShER</code> help</a></li>
        <li><a href="#a2-usher_to_taxonium-help">A.2 <code>usher_to_taxonium</code> help</a></li>
        <li><a href="#a3-fatovcf-help">A.3 <code>faToVcf</code> help</a></li>
      </ul>
    </li>
  </ul>
</nav>
	</div>
</div><div class="content post__content clearfix">
			<p>There are nearly 14 million viral genome sequences right now in the GISAID EpiCoV ™ database. It is not likely  to infer the phylogenetic relationships for such a huge dataset by traditional maximum likelyhood or Bayesian methods in a shor time period. The UShER package was developed to generate ultra-large phylogenetic tree of SARS-CoV-2 genomes. The algorithm of the UShER program is to place new samples onto an existing phylogeny using maximum parsimony method. It is able to place given SARS-CoV-2 genome sequences into the GISAID global phylogeny in a couple of hours.  This program is particularly helpful in understanding the relationships of newly sequenced SARS-CoV-2 genomes with each other and with previously sequenced genomes in a global phylogeny.</p>
<p>The UShER package is composed by four programs:</p>
<ol>
<li><code>UShER</code>: a program that rapidly places new samples onto an existing phylogeny using maximum parsimony.</li>
<li><code>matUtils</code>:  a toolkit for querying, interpreting and manipulating the mutation-annotated trees (MATs).</li>
<li><code>matOptimize</code>:  a program to rapidly and effectively optimize a mutation-annotated tree (MAT) for parsimony using subtree pruning and regrafting (SPR) moves within a user-defined radius.</li>
<li><code>RIPPLES</code>: a program that uses a phylogenomic technique to rapidly and sensitively detect recombinant nodes and their ancestors in a mutation-annotated tree (MAT).</li>
</ol>
<p>The taxoniumtools and <a href="https://taxonium.org/">Taxonium website</a> are used to display the MAT generated by UShER.</p>
<h1 id="1-installation">1. Installation</h1>
<h2 id="11-install-usher-package-via-conda">1.1 Install UShER package via conda</h2>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-bash" data-lang="bash"><span style="display:flex;"><span><span style="color:#75715e"># Create a new environment for UShER</span>
</span></span><span style="display:flex;"><span>conda create -n usher
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Activate the newly created environment</span>
</span></span><span style="display:flex;"><span>conda activate usher
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Set up channels</span>
</span></span><span style="display:flex;"><span>conda config --add channels defaults
</span></span><span style="display:flex;"><span>conda config --add channels bioconda
</span></span><span style="display:flex;"><span>conda config --add channels conda-forge
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Install the UShER package</span>
</span></span><span style="display:flex;"><span>conda install usher
</span></span></code></pre></div><h2 id="12-install-taxoniumtools">1.2 Install taxoniumtools</h2>
<p>Although the UShER wiki recommends <code>matUtils extract</code> to produce a MAT for Taxonium visualization, Taxonium recommends <code>taxoniumtools</code> to do this conversion for current Taxonium Version 2.</p>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-bash" data-lang="bash"><span style="display:flex;"><span><span style="color:#75715e"># If Python 3.x were not installed in the usher environment</span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># It will also install pip</span>
</span></span><span style="display:flex;"><span>conda install -c conda-forge python
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Install taxoniumtools</span>
</span></span><span style="display:flex;"><span>pip install taxoniumtools
</span></span></code></pre></div><h1 id="2-procedures">2. Procedures</h1>
<h2 id="21-download-sequences-and-metadata">2.1 Download sequences and metadata</h2>
<p>Download public tree (pb format), sample sequences, reference genome, and sites to mask:</p>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-bash" data-lang="bash"><span style="display:flex;"><span><span style="color:#75715e"># Protocol buffer format Mutation Annotated Tree, generated at 2021-08-04</span>
</span></span><span style="display:flex;"><span>wget http://hgdownload.soe.ucsc.edu/goldenPath/wuhCor1/UShER_SARS-CoV-2/2021/08/04/public-2021-08-04.all.masked.pb
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Metadata</span>
</span></span><span style="display:flex;"><span>wget https://hgdownload.soe.ucsc.edu/goldenPath/wuhCor1/UShER_SARS-CoV-2/2021/08/04/public-2021-08-04.metadata.tsv.gz
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># 15 sample sequences, S1 ~ S15</span>
</span></span><span style="display:flex;"><span>wget https://raw.githubusercontent.com/bpt26/usher_wiki/main/docs/source/test_samples.fa
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Reference genome, NC_045512v2</span>
</span></span><span style="display:flex;"><span>wget https://hgdownload.soe.ucsc.edu/goldenPath/wuhCor1/bigZips/wuhCor1.fa.gz
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># GenBank format format annotation file of reference genome</span>
</span></span><span style="display:flex;"><span>wget https://raw.githubusercontent.com/theosanderson/taxonium/master/taxoniumtools/test_data/hu1.gb
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span><span style="color:#75715e"># Problem sites, will be masked</span>
</span></span><span style="display:flex;"><span>wget https://raw.githubusercontent.com/W-L/ProblematicSites_SARS-CoV2/master/problematic_sites_sarsCov2.vcf
</span></span></code></pre></div><blockquote>
<p><strong>Note:</strong></p>
<ul>
<li>The <em><strong>latest</strong></em> protcol buffer format Mutation Annotated Tree (MAT) is available at:
<code>http://hgdownload.soe.ucsc.edu/goldenPath/wuhCor1/UShER_SARS-CoV-2/public-latest.all.masked.pb.gz</code></li>
<li>This tree includes only public sequences aggregated from GenBank, COG-UK, and the
China National Center for Bioinformation. Please read more details at:</li>
</ul>
<ul>
<li><a href="https://hgdownload.soe.ucsc.edu/goldenPath/wuhCor1/UShER_SARS-CoV-2/">https://hgdownload.soe.ucsc.edu/goldenPath/wuhCor1/UShER_SARS-CoV-2/</a></li>
</ul>
</blockquote>
<h2 id="22-operation">2.2 Operation</h2>
<ol>
<li>Unzip reference sequence:</li>
</ol>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-bash" data-lang="bash"><span style="display:flex;"><span>gunzip wuhCor1.fa.gz
</span></span></code></pre></div><ol start="2">
<li>Alignment sample sequences against reference genome by MAFFT</li>
</ol>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-bash" data-lang="bash"><span style="display:flex;"><span>mafft --thread <span style="color:#ae81ff">10</span> --auto --keeplength --addfragments test_samples.fa wuhCor1.fa &gt; aligned_seqs.fa
</span></span></code></pre></div><blockquote>
<p><strong>Note:</strong></p>
</blockquote>
<blockquote>
<p>MAFFT options:</p>
<ul>
<li><code>--auto</code>: automatically switches algorithm according to data size.  Safer to always use this flag.</li>
<li><code>--keeplength</code>: the alignment length is unchanged.  Insertions at the fragmentary sequences are deleted.</li>
<li><code>--addfragments</code>: Adding unaligned fragmentary sequence(s) into an existing alignment.</li>
</ul>
</blockquote>
<ol start="3">
<li>Convert above FASTA format alignment file to Variant Call Format (VCF) single-nucleotide diffs</li>
</ol>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-bash" data-lang="bash"><span style="display:flex;"><span>faToVcf -maskSites<span style="color:#f92672">=</span>problematic_sites_sarsCov2.vcf aligned_seqs.fa aligned_seqs.vcf
</span></span></code></pre></div><ol start="4">
<li>Now, add test samples to the protobuf MAT:</li>
</ol>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-bash" data-lang="bash"><span style="display:flex;"><span>usher -T <span style="color:#ae81ff">10</span> -i public-latest.all.masked.pb.gz -v aligned_seqs.vcf -o user_seqs.pb
</span></span></code></pre></div><ol start="5">
<li>Convert UShER <code>pb</code> format MAT to Taxonium <code>jsnl</code> format:</li>
</ol>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-bash" data-lang="bash"><span style="display:flex;"><span>usher_to_taxonium --input user_seqs.pb --output user_seqs-taxonium.jsonl.gz --metadata public-2021-08-04.metadata.tsv.gz --genbank hu1.gb --columns genbank_accession,country,date,pangolin_lineage
</span></span></code></pre></div><blockquote>
<p><strong>Note:</strong>
Do not download gene annotation file from NC_045512.2 GenBank record directly, use the optimized annotation file <code>hu1.gb</code> from taxonium.</p>
</blockquote>
<h2 id="23-tree-visualization">2.3 Tree visualization</h2>
<p>Open the generated <code>user_seqs-taxonium.jsonl.gz</code> tree file at Taxonium <a href="https://taxonium.org/">website</a>.</p>
<p><img src="https://usher-wiki.readthedocs.io/en/latest/_images/taxodium.jpeg" alt="Sample tree"></p>
<p>Please see the <a href="https://docs.taxonium.org/en/latest/">Taxonium documentation</a> for usage details.</p>
<h1 id="references">References</h1>
<ol>
<li><a href="https://usher-wiki.readthedocs.io/en/latest/">UShER Wiki</a></li>
<li><a href="https://docs.taxonium.org/en/latest/">Taxonium documentation</a></li>
<li><a href="https://github.com/W-L/ProblematicSites_SARS-CoV2">Problematic sites of SARS-CoV-2 sequence alignment</a></li>
</ol>
<h1 id="appendix-a-command-line-options">Appendix A. Command line options</h1>
<h2 id="a1-usher-help">A.1 <code>UShER</code> help</h2>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-bash" data-lang="bash"><span style="display:flex;"><span>UShER <span style="color:#f92672">(</span>v0.6.0<span style="color:#f92672">)</span>
</span></span><span style="display:flex;"><span>Options:
</span></span><span style="display:flex;"><span>  -v <span style="color:#f92672">[</span> --vcf <span style="color:#f92672">]</span> arg                      Input VCF file <span style="color:#f92672">(</span>in uncompressed or 
</span></span><span style="display:flex;"><span>                                        gzip-compressed .gz format<span style="color:#f92672">)</span> <span style="color:#f92672">[</span>REQUIRED<span style="color:#f92672">]</span>
</span></span><span style="display:flex;"><span>  -t <span style="color:#f92672">[</span> --tree <span style="color:#f92672">]</span> arg                     Input tree file
</span></span><span style="display:flex;"><span>  -d <span style="color:#f92672">[</span> --outdir <span style="color:#f92672">]</span> arg <span style="color:#f92672">(=</span>.<span style="color:#f92672">)</span>              Output directory to dump output and log
</span></span><span style="display:flex;"><span>                                        files <span style="color:#f92672">[</span>DEFAULT uses current directory<span style="color:#f92672">]</span>
</span></span><span style="display:flex;"><span>  -i <span style="color:#f92672">[</span> --load-mutation-annotated-tree <span style="color:#f92672">]</span> arg
</span></span><span style="display:flex;"><span>                                        Load mutation-annotated tree object
</span></span><span style="display:flex;"><span>  -o <span style="color:#f92672">[</span> --save-mutation-annotated-tree <span style="color:#f92672">]</span> arg
</span></span><span style="display:flex;"><span>                                        Save output mutation-annotated tree 
</span></span><span style="display:flex;"><span>                                        object to the specified filename
</span></span><span style="display:flex;"><span>  -s <span style="color:#f92672">[</span> --sort-before-placement-1 <span style="color:#f92672">]</span>      Sort new samples based on computed 
</span></span><span style="display:flex;"><span>                                        parsimony score and <span style="color:#66d9ef">then</span> number of 
</span></span><span style="display:flex;"><span>                                        optimal placements before the actual 
</span></span><span style="display:flex;"><span>                                        placement <span style="color:#f92672">[</span>EXPERIMENTAL<span style="color:#f92672">]</span>.
</span></span><span style="display:flex;"><span>  -S <span style="color:#f92672">[</span> --sort-before-placement-2 <span style="color:#f92672">]</span>      Sort new samples based on the number of
</span></span><span style="display:flex;"><span>                                        optimal placements and <span style="color:#66d9ef">then</span> the 
</span></span><span style="display:flex;"><span>                                        parsimony score before the actual 
</span></span><span style="display:flex;"><span>                                        placement <span style="color:#f92672">[</span>EXPERIMENTAL<span style="color:#f92672">]</span>.
</span></span><span style="display:flex;"><span>  -A <span style="color:#f92672">[</span> --sort-before-placement-3 <span style="color:#f92672">]</span>      Sort new samples based on the number of
</span></span><span style="display:flex;"><span>                                        ambiguous bases <span style="color:#f92672">[</span>EXPERIMENTAL<span style="color:#f92672">]</span>.
</span></span><span style="display:flex;"><span>  -r <span style="color:#f92672">[</span> --reverse-sort <span style="color:#f92672">]</span>                 Reverse the sorting order of sorting 
</span></span><span style="display:flex;"><span>                                        options <span style="color:#f92672">(</span>sort-before-placement-1 or 
</span></span><span style="display:flex;"><span>                                        sort-before-placement-2<span style="color:#f92672">)</span> <span style="color:#f92672">[</span>EXPERIMENTAL<span style="color:#f92672">]</span>
</span></span><span style="display:flex;"><span>  -c <span style="color:#f92672">[</span> --collapse-tree <span style="color:#f92672">]</span>                Collapse internal nodes of the input 
</span></span><span style="display:flex;"><span>                                        tree with no mutations and condense 
</span></span><span style="display:flex;"><span>                                        identical sequences in polytomies into 
</span></span><span style="display:flex;"><span>                                        a single node and the save the tree to 
</span></span><span style="display:flex;"><span>                                        file condensed-tree.nh in outdir
</span></span><span style="display:flex;"><span>  -C <span style="color:#f92672">[</span> --collapse-output-tree <span style="color:#f92672">]</span>         Collapse internal nodes of the output 
</span></span><span style="display:flex;"><span>                                        tree with no mutations before the 
</span></span><span style="display:flex;"><span>                                        saving the tree to file final-tree.nh 
</span></span><span style="display:flex;"><span>                                        in outdir
</span></span><span style="display:flex;"><span>  -e <span style="color:#f92672">[</span> --max-uncertainty-per-sample <span style="color:#f92672">]</span> arg <span style="color:#f92672">(=</span>1000000<span style="color:#f92672">)</span>
</span></span><span style="display:flex;"><span>                                        Maximum number of equally parsimonious 
</span></span><span style="display:flex;"><span>                                        placements allowed per sample beyond 
</span></span><span style="display:flex;"><span>                                        which the sample is ignored
</span></span><span style="display:flex;"><span>  -E <span style="color:#f92672">[</span> --max-parsimony-per-sample <span style="color:#f92672">]</span> arg <span style="color:#f92672">(=</span>1000000<span style="color:#f92672">)</span>
</span></span><span style="display:flex;"><span>                                        Maximum parsimony score of the most 
</span></span><span style="display:flex;"><span>                                        parsimonious placement<span style="color:#f92672">(</span>s<span style="color:#f92672">)</span> allowed per 
</span></span><span style="display:flex;"><span>                                        sample beyond which the sample is 
</span></span><span style="display:flex;"><span>                                        ignored
</span></span><span style="display:flex;"><span>  -u <span style="color:#f92672">[</span> --write-uncondensed-final-tree <span style="color:#f92672">]</span> 
</span></span><span style="display:flex;"><span>                                        Write the final tree in uncondensed 
</span></span><span style="display:flex;"><span>                                        format and save to file 
</span></span><span style="display:flex;"><span>                                        uncondensed-final-tree.nh in outdir
</span></span><span style="display:flex;"><span>  -k <span style="color:#f92672">[</span> --write-subtrees-size <span style="color:#f92672">]</span> arg <span style="color:#f92672">(=</span>0<span style="color:#f92672">)</span> Write minimum set of subtrees covering 
</span></span><span style="display:flex;"><span>                                        the newly added samples of size equal 
</span></span><span style="display:flex;"><span>                                        to this value
</span></span><span style="display:flex;"><span>  -K <span style="color:#f92672">[</span> --write-single-subtree <span style="color:#f92672">]</span> arg <span style="color:#f92672">(=</span>0<span style="color:#f92672">)</span>
</span></span><span style="display:flex;"><span>                                        Similar to write-subtrees-size but 
</span></span><span style="display:flex;"><span>                                        produces a single subtree with all 
</span></span><span style="display:flex;"><span>                                        newly added samples along with random 
</span></span><span style="display:flex;"><span>                                        samples up to the value specified by 
</span></span><span style="display:flex;"><span>                                        this argument
</span></span><span style="display:flex;"><span>  -p <span style="color:#f92672">[</span> --write-parsimony-scores-per-node <span style="color:#f92672">]</span> 
</span></span><span style="display:flex;"><span>                                        Write the parsimony scores <span style="color:#66d9ef">for</span> adding 
</span></span><span style="display:flex;"><span>                                        new samples at each existing node in 
</span></span><span style="display:flex;"><span>                                        the tree without modifying the tree in 
</span></span><span style="display:flex;"><span>                                        a file names parsimony-scores.tsv in 
</span></span><span style="display:flex;"><span>                                        outdir
</span></span><span style="display:flex;"><span>  -M <span style="color:#f92672">[</span> --multiple-placements <span style="color:#f92672">]</span> arg <span style="color:#f92672">(=</span>1<span style="color:#f92672">)</span> Create a new tree up to this limit <span style="color:#66d9ef">for</span> 
</span></span><span style="display:flex;"><span>                                        each possibility of parsimony-optimal 
</span></span><span style="display:flex;"><span>                                        placement
</span></span><span style="display:flex;"><span>  -l <span style="color:#f92672">[</span> --retain-input-branch-lengths <span style="color:#f92672">]</span>  Retain the branch lengths from the 
</span></span><span style="display:flex;"><span>                                        input tree in out newick files instead 
</span></span><span style="display:flex;"><span>                                        of using number of mutations <span style="color:#66d9ef">for</span> the 
</span></span><span style="display:flex;"><span>                                        branch lengths.
</span></span><span style="display:flex;"><span>  -n <span style="color:#f92672">[</span> --no-add <span style="color:#f92672">]</span>                       Do not add new samples to the tree
</span></span><span style="display:flex;"><span>  -D <span style="color:#f92672">[</span> --detailed-clades <span style="color:#f92672">]</span>              In clades.txt, write a histogram of 
</span></span><span style="display:flex;"><span>                                        annotated clades and counts across all 
</span></span><span style="display:flex;"><span>                                        equally parsimonious placements
</span></span><span style="display:flex;"><span>  -T <span style="color:#f92672">[</span> --threads <span style="color:#f92672">]</span> arg <span style="color:#f92672">(=</span>104<span style="color:#f92672">)</span>           Number of threads to use when possible 
</span></span><span style="display:flex;"><span>                                        <span style="color:#f92672">[</span>DEFAULT uses all available cores, <span style="color:#ae81ff">104</span> 
</span></span><span style="display:flex;"><span>                                        detected on this machine<span style="color:#f92672">]</span>
</span></span><span style="display:flex;"><span>  --version                             Print version number
</span></span><span style="display:flex;"><span>  -h <span style="color:#f92672">[</span> --help <span style="color:#f92672">]</span>                         Print help messages
</span></span></code></pre></div><h2 id="a2-usher_to_taxonium-help">A.2 <code>usher_to_taxonium</code> help</h2>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-bash" data-lang="bash"><span style="display:flex;"><span>usage: usher_to_taxonium <span style="color:#f92672">[</span>-h<span style="color:#f92672">]</span> -i INPUT -o OUTPUT <span style="color:#f92672">[</span>-m METADATA<span style="color:#f92672">]</span> <span style="color:#f92672">[</span>-g GENBANK<span style="color:#f92672">]</span>
</span></span><span style="display:flex;"><span>                         <span style="color:#f92672">[</span>-c COLUMNS<span style="color:#f92672">]</span> <span style="color:#f92672">[</span>-C<span style="color:#f92672">]</span>
</span></span><span style="display:flex;"><span>                         <span style="color:#f92672">[</span>--chronumental_steps CHRONUMENTAL_STEPS<span style="color:#f92672">]</span>
</span></span><span style="display:flex;"><span>                         <span style="color:#f92672">[</span>--chronumental_date_output CHRONUMENTAL_DATE_OUTPUT<span style="color:#f92672">]</span>
</span></span><span style="display:flex;"><span>                         <span style="color:#f92672">[</span>--chronumental_tree_output CHRONUMENTAL_TREE_OUTPUT<span style="color:#f92672">]</span>
</span></span><span style="display:flex;"><span>                         <span style="color:#f92672">[</span>--chronumental_reference_node CHRONUMENTAL_REFERENCE_NODE<span style="color:#f92672">]</span>
</span></span><span style="display:flex;"><span>                         <span style="color:#f92672">[</span>-j CONFIG_JSON<span style="color:#f92672">]</span> <span style="color:#f92672">[</span>-t TITLE<span style="color:#f92672">]</span>
</span></span><span style="display:flex;"><span>                         <span style="color:#f92672">[</span>--overlay_html OVERLAY_HTML<span style="color:#f92672">]</span> <span style="color:#f92672">[</span>--remove_after_pipe<span style="color:#f92672">]</span>
</span></span><span style="display:flex;"><span>                         <span style="color:#f92672">[</span>--clade_types CLADE_TYPES<span style="color:#f92672">]</span> <span style="color:#f92672">[</span>--name_internal_nodes<span style="color:#f92672">]</span>
</span></span><span style="display:flex;"><span>                         <span style="color:#f92672">[</span>--shear<span style="color:#f92672">]</span> <span style="color:#f92672">[</span>--shear_threshold SHEAR_THRESHOLD<span style="color:#f92672">]</span>
</span></span><span style="display:flex;"><span>                         <span style="color:#f92672">[</span>--only_variable_sites<span style="color:#f92672">]</span> <span style="color:#f92672">[</span>--key_column KEY_COLUMN<span style="color:#f92672">]</span>
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span>Convert a Usher pb to Taxonium jsonl format
</span></span><span style="display:flex;"><span>
</span></span><span style="display:flex;"><span>options:
</span></span><span style="display:flex;"><span>  -h, --help            show this help message and exit
</span></span><span style="display:flex;"><span>  -i INPUT, --input INPUT
</span></span><span style="display:flex;"><span>                        File path to input Usher protobuf file <span style="color:#f92672">(</span>.pb<span style="color:#f92672">)</span>
</span></span><span style="display:flex;"><span>  -o OUTPUT, --output OUTPUT
</span></span><span style="display:flex;"><span>                        File path <span style="color:#66d9ef">for</span> output Taxonium jsonl file
</span></span><span style="display:flex;"><span>  -m METADATA, --metadata METADATA
</span></span><span style="display:flex;"><span>                        File path <span style="color:#66d9ef">for</span> input metadata file <span style="color:#f92672">(</span>CSV/TSV<span style="color:#f92672">)</span>
</span></span><span style="display:flex;"><span>  -g GENBANK, --genbank GENBANK
</span></span><span style="display:flex;"><span>                        File path <span style="color:#66d9ef">for</span> GenBank file containing reference genome
</span></span><span style="display:flex;"><span>                        <span style="color:#f92672">(</span>N.B. currently only one chromosome is supported, and
</span></span><span style="display:flex;"><span>                        no compound features<span style="color:#f92672">)</span>
</span></span><span style="display:flex;"><span>  -c COLUMNS, --columns COLUMNS
</span></span><span style="display:flex;"><span>                        Column names to include in the metadata, separated by
</span></span><span style="display:flex;"><span>                        commas, e.g. <span style="color:#e6db74">`</span>pangolin_lineage,country<span style="color:#e6db74">`</span>
</span></span><span style="display:flex;"><span>  -C, --chronumental    Runs Chronumental to build a time tree. The metadata
</span></span><span style="display:flex;"><span>                        TSV must include a date column.
</span></span><span style="display:flex;"><span>  --chronumental_steps CHRONUMENTAL_STEPS
</span></span><span style="display:flex;"><span>                        Number of steps to run Chronumental <span style="color:#66d9ef">for</span>
</span></span><span style="display:flex;"><span>  --chronumental_date_output CHRONUMENTAL_DATE_OUTPUT
</span></span><span style="display:flex;"><span>                        Optional output file <span style="color:#66d9ef">for</span> the chronumental date table
</span></span><span style="display:flex;"><span>                        <span style="color:#66d9ef">if</span> you want to keep it <span style="color:#f92672">(</span>a table mapping nodes to their
</span></span><span style="display:flex;"><span>                        inferred dates<span style="color:#f92672">)</span>.
</span></span><span style="display:flex;"><span>  --chronumental_tree_output CHRONUMENTAL_TREE_OUTPUT
</span></span><span style="display:flex;"><span>                        Optional output file <span style="color:#66d9ef">for</span> the chronumental time tree
</span></span><span style="display:flex;"><span>                        file in nwk format.
</span></span><span style="display:flex;"><span>  --chronumental_reference_node CHRONUMENTAL_REFERENCE_NODE
</span></span><span style="display:flex;"><span>                        A reference node to be used <span style="color:#66d9ef">for</span> Chronumental. This
</span></span><span style="display:flex;"><span>                        should be earlier in the outbreak and have a good
</span></span><span style="display:flex;"><span>                        defined date. If not set the oldest sample will be
</span></span><span style="display:flex;"><span>                        automatically picked by Chronumental.
</span></span><span style="display:flex;"><span>  -j CONFIG_JSON, --config_json CONFIG_JSON
</span></span><span style="display:flex;"><span>                        A JSON file to use as a config file containing things
</span></span><span style="display:flex;"><span>                        such as search parameters
</span></span><span style="display:flex;"><span>  -t TITLE, --title TITLE
</span></span><span style="display:flex;"><span>                        A title <span style="color:#66d9ef">for</span> the tree. This will be shown at the top of
</span></span><span style="display:flex;"><span>                        the window as <span style="color:#e6db74">&#34;[Title] - powered by Taxonium&#34;</span>
</span></span><span style="display:flex;"><span>  --overlay_html OVERLAY_HTML
</span></span><span style="display:flex;"><span>                        A file containing HTML to put in the About box when
</span></span><span style="display:flex;"><span>                        this tree is loaded. This could contain information
</span></span><span style="display:flex;"><span>                        about who built the tree and what data you used.
</span></span><span style="display:flex;"><span>  --remove_after_pipe   If set, we will remove anything after a pipe <span style="color:#f92672">(</span>|<span style="color:#f92672">)</span> in
</span></span><span style="display:flex;"><span>                        each node<span style="color:#e6db74">&#39;s name, after joining to metadata
</span></span></span><span style="display:flex;"><span><span style="color:#e6db74">  --clade_types CLADE_TYPES
</span></span></span><span style="display:flex;"><span><span style="color:#e6db74">                        Optionally specify clade types provided in the UShER
</span></span></span><span style="display:flex;"><span><span style="color:#e6db74">                        file, comma separated - e.g. &#39;</span>nextstrain,pango<span style="color:#e6db74">&#39;. Order
</span></span></span><span style="display:flex;"><span><span style="color:#e6db74">                        must match that used in the UShER pb file. If you
</span></span></span><span style="display:flex;"><span><span style="color:#e6db74">                        haven&#39;</span>t specifically annotated clades in your
</span></span><span style="display:flex;"><span>                        protobuf, don<span style="color:#e6db74">&#39;t use this
</span></span></span><span style="display:flex;"><span><span style="color:#e6db74">  --name_internal_nodes
</span></span></span><span style="display:flex;"><span><span style="color:#e6db74">                        If set, we will name internal nodes node_xxx
</span></span></span><span style="display:flex;"><span><span style="color:#e6db74">  --shear               If set, we will &#39;</span>shear<span style="color:#960050;background-color:#1e0010">&#39;</span> the tree. This will iterate
</span></span><span style="display:flex;"><span>                        over all nodes. If a particular sub-branch makes up
</span></span><span style="display:flex;"><span>                        fewer than e.g. 1/1000 of the total descendants, <span style="color:#66d9ef">then</span>
</span></span><span style="display:flex;"><span>                        in most cases it represents a sequencing error. <span style="color:#f92672">(</span>But
</span></span><span style="display:flex;"><span>                        it also could represent recombinants, or a real, unfit
</span></span><span style="display:flex;"><span>                        branch.<span style="color:#f92672">)</span> We remove these to simplify the
</span></span><span style="display:flex;"><span>                        interpretation of the tree.
</span></span><span style="display:flex;"><span>  --shear_threshold SHEAR_THRESHOLD
</span></span><span style="display:flex;"><span>                        Threshold <span style="color:#66d9ef">for</span> shearing, default is <span style="color:#ae81ff">1000</span> meaning
</span></span><span style="display:flex;"><span>                        branches will be removed <span style="color:#66d9ef">if</span> they make up less than
</span></span><span style="display:flex;"><span>                        &lt;1/1000 nodes. Has no effect unless --shear is set.
</span></span><span style="display:flex;"><span>  --only_variable_sites
</span></span><span style="display:flex;"><span>                        Only store information about the root sequence at a
</span></span><span style="display:flex;"><span>                        particular position <span style="color:#66d9ef">if</span> there is variation at that
</span></span><span style="display:flex;"><span>                        position somewhere in the tree. This helps to speed up
</span></span><span style="display:flex;"><span>                        the loading of larger genomes such as MPXV.
</span></span><span style="display:flex;"><span>  --key_column KEY_COLUMN
</span></span><span style="display:flex;"><span>                        The column in the metadata file which is the same as
</span></span><span style="display:flex;"><span>                        the names in the tree
</span></span></code></pre></div><h2 id="a3-fatovcf-help">A.3 <code>faToVcf</code> help</h2>
<div class="highlight"><pre tabindex="0" style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4;"><code class="language-bash" data-lang="bash"><span style="display:flex;"><span>faToVcf - Convert a FASTA alignment file to Variant Call Format <span style="color:#f92672">(</span>VCF<span style="color:#f92672">)</span> single-nucleotide diffs
</span></span><span style="display:flex;"><span>usage:
</span></span><span style="display:flex;"><span>   faToVcf in.fa out.vcf
</span></span><span style="display:flex;"><span>options:
</span></span><span style="display:flex;"><span>   -ambiguousToN         Treat all IUPAC ambiguous bases <span style="color:#f92672">(</span>N, R, V etc<span style="color:#f92672">)</span> as N <span style="color:#f92672">(</span>no call<span style="color:#f92672">)</span>.
</span></span><span style="display:flex;"><span>   -excludeFile<span style="color:#f92672">=</span>file     Exclude sequences named in file which has one sequence name per line
</span></span><span style="display:flex;"><span>   -includeNoAltN        Include base positions with no alternate alleles observed, but at
</span></span><span style="display:flex;"><span>                         least one N <span style="color:#f92672">(</span>missing base / no-call<span style="color:#f92672">)</span>
</span></span><span style="display:flex;"><span>   -includeRef           Include the reference in the genotype columns
</span></span><span style="display:flex;"><span>                         <span style="color:#f92672">(</span>default: omitted as redundant<span style="color:#f92672">)</span>
</span></span><span style="display:flex;"><span>   -maskSites<span style="color:#f92672">=</span>file       Exclude variants in positions recommended <span style="color:#66d9ef">for</span> masking in file
</span></span><span style="display:flex;"><span>                         <span style="color:#f92672">(</span>typically https://github.com/W-L/ProblematicSites_SARS-CoV2/raw/master/problematic_sites_sarsCov2.vcf<span style="color:#f92672">)</span>
</span></span><span style="display:flex;"><span>   -maxDiff<span style="color:#f92672">=</span>N            Exclude sequences with more than N mismatches with the reference
</span></span><span style="display:flex;"><span>                         <span style="color:#f92672">(</span><span style="color:#66d9ef">if</span> -windowSize is used, sequences are masked accordingly first<span style="color:#f92672">)</span>
</span></span><span style="display:flex;"><span>   -minAc<span style="color:#f92672">=</span>N              Ignore alternate alleles observed fewer than N times
</span></span><span style="display:flex;"><span>   -minAf<span style="color:#f92672">=</span>F              Ignore alternate alleles observed in less than F of non-N bases
</span></span><span style="display:flex;"><span>   -minAmbigInWindow<span style="color:#f92672">=</span>N   When -windowSize is provided, mask any base <span style="color:#66d9ef">for</span> which there are at
</span></span><span style="display:flex;"><span>                         least this many N, ambiguous or gap characters within the window.
</span></span><span style="display:flex;"><span>                         <span style="color:#f92672">(</span>default: 2<span style="color:#f92672">)</span>
</span></span><span style="display:flex;"><span>   -noGenotypes          Output 8-column VCF, without the sample genotype columns
</span></span><span style="display:flex;"><span>   -ref<span style="color:#f92672">=</span>seqName          Use seqName as the reference sequence; must be present in faFile
</span></span><span style="display:flex;"><span>                         <span style="color:#f92672">(</span>default: first sequence in faFile<span style="color:#f92672">)</span>
</span></span><span style="display:flex;"><span>   -resolveAmbiguous     For IUPAC ambiguous characters like R <span style="color:#f92672">(</span>A or G<span style="color:#f92672">)</span>, <span style="color:#66d9ef">if</span> the character
</span></span><span style="display:flex;"><span>                         represents two bases and one is the reference base, convert it to the
</span></span><span style="display:flex;"><span>                         non-reference base.  Otherwise convert it to N.
</span></span><span style="display:flex;"><span>   -startOffset<span style="color:#f92672">=</span>N        Add N bases to each position <span style="color:#f92672">(</span><span style="color:#66d9ef">for</span> trimmed alignments<span style="color:#f92672">)</span>
</span></span><span style="display:flex;"><span>   -vcfChrom<span style="color:#f92672">=</span>seqName     Use seqName <span style="color:#66d9ef">for</span> the CHROM column in VCF <span style="color:#f92672">(</span>default: ref sequence<span style="color:#f92672">)</span>
</span></span><span style="display:flex;"><span>   -windowSize<span style="color:#f92672">=</span>N         Mask any base <span style="color:#66d9ef">for</span> which there are at least -minAmbigWindow bases in a
</span></span><span style="display:flex;"><span>                         window of +-N bases around the base.  Masking approach adapted from
</span></span><span style="display:flex;"><span>                         https://github.com/roblanf/sarscov2phylo/ file scripts/mask_seq.py
</span></span><span style="display:flex;"><span>                         Use -windowSize<span style="color:#f92672">=</span><span style="color:#ae81ff">7</span> <span style="color:#66d9ef">for</span> same results.
</span></span><span style="display:flex;"><span>in.fa must contain a series of sequences with different names and the same length.
</span></span><span style="display:flex;"><span>Both N and - are treated as missing information.
</span></span></code></pre></div>
		</div>
		<footer class="post__footer">
			
<div class="post__tags tags clearfix">
	<svg class="tags__badge icon icon-tag" width="16" height="16" viewBox="0 0 32 32"><path d="M32 19c0 1-1 2-1 2L21 31s-1 1-2 1-2-1-2-1L2 16c-1-1-1.4-2-1.4-2S0 12.5 0 11V3C0 1.5.8.8.8.8S1.5 0 3 0h8c1.5 0 3 .6 3 .6S15 1 16 2l15 15s1 1 1 2zM7 10a3 3 0 1 0 0-6 3 3 0 0 0 0 6z"/></svg>
	<ul class="tags__list">
		<li class="tags__item">
			<a class="tags__link btn" href="/tags/tree/" rel="tag">Tree</a>
		</li>
		<li class="tags__item">
			<a class="tags__link btn" href="/tags/usher/" rel="tag">UShER</a>
		</li>
		<li class="tags__item">
			<a class="tags__link btn" href="/tags/taxonium/" rel="tag">Taxonium</a>
		</li>
	</ul>
</div>
		</footer>
	</article>
</main>


<nav class="pager flex">
	<div class="pager__item pager__item--prev">
		<a class="pager__link" href="/posts/2022-11-07-check-and-upgrade-bioconductor-packages/" rel="prev">
			<span class="pager__subtitle">«&thinsp;Previous</span>
			<p class="pager__title">Check and Upgrade Bioconductor Packages</p>
		</a>
	</div>
	<div class="pager__item pager__item--next">
		<a class="pager__link" href="/posts/2022-11-25-alignment-of-sars-cov-2-reference-genome-sequences/" rel="next">
			<span class="pager__subtitle">Next&thinsp;»</span>
			<p class="pager__title">Alignment of SARS-CoV-2 Reference Genome Sequences</p>
		</a>
	</div>
</nav>

  <article class="post">
    <script 
        data-isso="https://bvlu.602204.xyz/isso/" 
        data-isso-id="thread-id"
        data-isso-css="true"
        data-isso-lang="en"
        data-isso-reply-to-self="true"
        data-isso-require-author="true" 
        data-isso-require-email="false" 
        data-isso-avatar="true"
        data-isso-avatar-bg=""
        src="https://bvlu.602204.xyz/isso/js/embed.min.js">
    </script>
    <noscript>Please enable JavaScript to view the comments powered by <a href="https://posativ.org/isso/">Isso</a>.</noscript>
    <div>
      <section id="isso-thread"></section>
    </div>
</article>



			</div>
			<aside class="sidebar"><div class="widget-search widget">
	<form class="widget-search__form" role="search" method="get" action="https://google.com/search">
		<label>
			<input class="widget-search__field" type="search" placeholder="SEARCH…" value="" name="q" aria-label="SEARCH…">
		</label>
		<input class="widget-search__submit" type="submit" value="Search">
		<input type="hidden" name="sitesearch" value="https://omicx.cc/" />
	</form>
</div>
<div class="widget-recent widget">
	<h4 class="widget__title">Recent Posts</h4>
	<div class="widget__content">
		<ul class="widget__list">
			<li class="widget__item"><a class="widget__link" href="/posts/2024-04-13-install-open-source-pymol-3-in-windows-11/">Install Open Source PyMOL 3.0 in Windows 11</a></li>
			<li class="widget__item"><a class="widget__link" href="/posts/2024-01-30-install-open-source-pymol-in-windows-11/">Install Open Source PyMOL in Windows 11</a></li>
			<li class="widget__item"><a class="widget__link" href="/posts/2024-01-11-vs-code-connect-to-remote-server-in-intranet/">VS Code Connect to Remote Server over Intranet</a></li>
			<li class="widget__item"><a class="widget__link" href="/posts/2023-12-25-install-bookstack-by-docker/">Install BookStack by Docker</a></li>
			<li class="widget__item"><a class="widget__link" href="/posts/2023-11-27-fix-cmake-unsupported-gnu-version-error/">Fix Cmake &#39;Unsupported GNU Version&#39; Error</a></li>
		</ul>
	</div>
</div>
<div class="widget-categories widget">
	<h4 class="widget__title">Categories</h4>
	<div class="widget__content">
		<ul class="widget__list">
			<li class="widget__item">
				<a class="widget__link" href="/categories/bd-rhapsody/">BD Rhapsody</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/beast/">BEAST</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/bioconductor/">Bioconductor</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/biosoft/">Biosoft</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/blog/">Blog</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/conda/">Conda</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/cuda/">CUDA</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/figtree/">FigTree</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/hardware/">Hardware</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/hugo/">Hugo</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/linux/">Linux</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/media/">Media</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/metagenomics/">Metagenomics</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/nas/">NAS</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/ngs/">NGS</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/note/">Note</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/phylogenetics/">Phylogenetics</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/programming/">Programming</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/pymol/">PyMOL</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/python/">Python</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/r/">R</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/raspberry-pi/">Raspberry Pi</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/scrna-seq/">ScRNA-Seq</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/software/">Software</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/sqlite/">SQLite</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/vnc/">VNC</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/website/">Website</a></li>
			<li class="widget__item">
				<a class="widget__link" href="/categories/xfce/">Xfce</a></li>
		</ul>
	</div>
</div>
<div class="widget-taglist widget">
	<h4 class="widget__title">Tags</h4>
	<div class="widget__content">
		<a class="widget-taglist__link widget__link btn" href="/tags/aria2/" title="Aria2">Aria2</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/bd-genomics/" title="BD Genomics">BD Genomics</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/bd-rhapsody/" title="BD Rhapsody">BD Rhapsody</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/beagle/" title="Beagle">Beagle</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/beast/" title="BEAST">BEAST</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/bioconductor/" title="Bioconductor">Bioconductor</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/bookstack/" title="Bookstack">Bookstack</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/certbot/" title="Certbot">Certbot</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/cloudflare/" title="Cloudflare">Cloudflare</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/conda/" title="Conda">Conda</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/cuda/" title="CUDA">CUDA</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/debian/" title="Debian">Debian</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/endnote/" title="EndNote">EndNote</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/flatpak/" title="Flatpak">Flatpak</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/fry/" title="Fry">Fry</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/gnu-parallel/" title="GNU Parallel">GNU Parallel</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/hugo/" title="Hugo">Hugo</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/jellyfin/" title="Jellyfin">Jellyfin</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/kvm/" title="KVM">KVM</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/letsencrypt/" title="Letsencrypt">Letsencrypt</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/linux/" title="Linux">Linux</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/mamba/" title="Mamba">Mamba</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/media/" title="Media">Media</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/metagenomics/" title="Metagenomics">Metagenomics</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/mrbayes/" title="MrBayes">MrBayes</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/nas/" title="NAS">NAS</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/needle/" title="Needle">Needle</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/nextstrain/" title="Nextstrain">Nextstrain</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/nginx/" title="Nginx">Nginx</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/ngs/" title="NGS">NGS</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/note/" title="Note">Note</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/nvidia/" title="Nvidia">Nvidia</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/onedrive/" title="OneDrive">OneDrive</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/pavian/" title="Pavian">Pavian</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/pymol/" title="PyMOL">PyMOL</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/qemu/" title="QEMU">QEMU</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/r/" title="R">R</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/r-markdown/" title="R Markdown">R Markdown</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/raspberry-pi/" title="Raspberry Pi">Raspberry Pi</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/rna-seq/" title="RNA-Seq">RNA-Seq</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/rstudio-server/" title="RStudio Server">RStudio Server</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/sars-cov-2/" title="SARS-CoV-2">SARS-CoV-2</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/scrna-seq/" title="ScRNA-Seq">ScRNA-Seq</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/sqlite/" title="SQLite">SQLite</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/taxonium/" title="Taxonium">Taxonium</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/tips/" title="Tips">Tips</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/tree/" title="Tree">Tree</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/usher/" title="UShER">UShER</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/vnc/" title="VNC">VNC</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/wsl/" title="WSL">WSL</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/xfce/" title="Xfce">Xfce</a>
		<a class="widget-taglist__link widget__link btn" href="/tags/zdock/" title="ZDOCK">ZDOCK</a>
	</div>
</div>
<div class="widget-social widget">
	<h4 class="widget-social__title widget__title">Social</h4>
	<div class="widget-social__content widget__content">
		<div class="widget-social__item widget__item">
			<a class="widget-social__link widget__link btn" title="GitHub" rel="noopener noreferrer" href="https://github.com/zer0liu" target="_blank">
				<svg class="widget-social__link-icon icon icon-github" width="24" height="24" viewBox="0 0 384 374"><path d="m192 0c-106.1 0-192 85.8-192 191.7 0 84.7 55 156.6 131.3 181.9 9.6 1.8 13.1-4.2 13.1-9.2 0-4.6-.2-16.6-.3-32.6-53.4 11.6-64.7-25.7-64.7-25.7-8.7-22.1-21.3-28-21.3-28-17.4-11.9 1.3-11.6 1.3-11.6 19.3 1.4 29.4 19.8 29.4 19.8 17.1 29.3 44.9 20.8 55.9 15.9 1.7-12.4 6.7-20.8 12.2-25.6-42.6-4.8-87.5-21.3-87.5-94.8 0-20.9 7.5-38 19.8-51.4-2-4.9-8.6-24.3 1.9-50.7 0 0 16.1-5.2 52.8 19.7 15.3-4.2 31.7-6.4 48.1-6.5 16.3.1 32.7 2.2 48.1 6.5 36.7-24.8 52.8-19.7 52.8-19.7 10.5 26.4 3.9 45.9 1.9 50.7 12.3 13.4 19.7 30.5 19.7 51.4 0 73.7-44.9 89.9-87.7 94.6 6.9 5.9 13 17.6 13 35.5 0 25.6-.2 46.3-.2 52.6 0 5.1 3.5 11.1 13.2 9.2 76.2-25.5 131.2-97.3 131.2-182 0-105.9-86-191.7-192-191.7z"/></svg>
				<span>GitHub</span>
			</a>
		</div>
		<div class="widget-social__item widget__item">
			<a class="widget-social__link widget__link btn" title="Email" href="mailto:omicx.cc@gmail.com">
				<svg class="widget-social__link-icon icon icon-mail" width="24" height="24" viewBox="0 0 416 288"><path d="m0 16v256 16h16 384 16v-16-256-16h-16-384-16zm347 16-139 92.5-139-92.5zm-148 125.5 9 5.5 9-5.5 167-111.5v210h-352v-210z"/></svg>
				<span>omicx.cc@gmail.com</span>
			</a>
		</div>

		
	</div>
</div>
</aside>
		</div>
		<footer class="footer">
	<div class="container footer__container flex">
		
		<div class="footer__copyright">
			&copy; 2024 omicx.cc.
			<span class="footer__copyright-credits">Generated with <a href="https://gohugo.io/" rel="nofollow noopener" target="_blank">Hugo</a> and <a href="https://github.com/Vimux/Mainroad/" rel="nofollow noopener" target="_blank">Mainroad</a> theme.</span>
		</div>
	</div>
</footer>

	</div>
<script async defer src="/js/menu.js"></script>
</body>
</html>
